### ISSUE TITLE
macOS: Building Development branch build fails

### ISSUE THREAD

**Issue by LB-MateuszSzweda** [o3de/o3de#6845]
**Describe the bug**
CMake fails to build development branch on macOS due to the following error:

```
** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution CMake\ PostBuild\ Rules /Users/macpro15/o3de/osx_build/Code/Tools/ProjectManager/O3DE.build/profile/ProjectManager.build/Script-B87D0F6DC34218257AFEB57C.sh (in target 'ProjectManager' from project 'O3DE')
(1 failure)
Elapsed time: 1016 s. (time), 0.063439 s. (clock)
```

The issue occured on every tested macOS machine with every tested project which were:

* Macbook Air 2020 (M1 AutomatedTesting, o3de-atom-sampleviewer)
* Macbook Pro 2017 (Intel i7 AutomatedTesting)

For further information please refer to the attached building logs.

**Steps to reproduce**

1. Obtain the Development branch build via Github.
2. Create a new or move your preexisting 3rdParty folder containing CMake into the o3de folder.
3. Obtain python packages with the _get_python.sh_ script located in the _o3de/python_ directory.
4. Create the osx_build folder under the o3de location.
5. Set the following temporary environment variables in the Terminal:
```
export PATH=$PATH:/Users/<home_folder_name>/o3de/3rdParty/CMake/CMake.app/Contents/bin
export LY_3RDPARTY_PATH=/Users/<home_folder_name>/o3de/3rdParty
export LY_PACKAGE_SERVER_URLS=https://d2c171ws20a1rv.cloudfront.net
```
6. Configure and generate the Xcode solution with the following command:
`cmake .. -G "Xcode" -DLY_3RDPARTY_PATH=$LY_3RDPARTY_PATH -DLY_PROJECTS=AutomatedTesting -DLY_UNITY_BUILD=1`

7. Build Mac binaries with the following command:
`cmake -E time cmake --build . --target ALL_BUILD --config profile`

**Expected behavior**
Building process finishes successfully without errors.

**Actual behavior**
Building fails.

**Found in Branch**
Development (44908a8)

**Desktop/Device**
- Device: Mac
- OS: macOS
- Version: Monterey 12.0.1
- CPU: M1
- GPU: M1
- Memory: 16GB

**Additional context**
[M1 macOS failed building log.txt](https://github.com/o3de/o3de/files/7854695/M1.macOS.failed.building.log.txt)
[i7 macOS failed building log.txt](https://github.com/o3de/o3de/files/7854696/i7.macOS.failed.building.log.txt)

**Comment by amzn-sj** [comment]
Fixed in https://github.com/o3de/o3de/pull/6857

**Comment by LB-MateuszSzweda** [comment]
Verified as fixed in Development (e224fc2). Project building succeeded on all macOS machines.
